javascript/code-challenges/challenges-06.test.js

```diff
@@ -0,0 +1,190 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named getNames that, given an array of people objects, uses map to return an array of names reversed.
+
+For example:
+[
+{
+  name: 'lloyd',
+  age: 32,
+  shoeSize: 12
+},
+{
+  name: 'jamie',
+  age: 21,
+  shoeSize: 8
+}
+]
+
+Returns: ['dyoll', 'eimaj'];
+------------------------------------------------------------------------------------------------ */
+
+const getNames = (arr) => {
+  // Solution code here...
+  return arr.map((item) => {
+    let reverseNames = "";
+    for (let i = item.name.length - 1; i >= 0; i--) {
+      reverseNames = reverseNames + item.name[i];
+    }
+    return reverseNames;
+  })
+};
```

Comment:
Hi there, good work on the challenges!

Feedback here: Consider whether you need the for loop (as we find solutions, it is helpful to go back and see if we can simplify the solution/improve the time complexity and maintainability of the code 👍 

 We can simplify this a bit as we know that using map() we are already iterating over the input array and modifying its elements using a callback function. So perhaps you can try something like this:
 
 ` arr.map(item => item.name` ---> after we can consider how we can reverse the items/strings inside (see split(), reverse(), and join() methods if you have time :)